### PR TITLE
added tests for ETL

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

**Comment by Giulio2002** [comment]
#122

File: db/silkworm/etl/collector_test.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    TemporaryDirectory tmp_dir;
+    // Initialize random seed
+    srand(time(NULL));
+    // Initialize temporary Database
+    lmdb::DatabaseConfig db_config{tmp_dir.path(), 32 * kMebi};
+    db_config.set_readonly(false);
+    auto env{lmdb::get_env(db_config)};
+    auto txn{env->begin_rw_transaction()};
+    // Create temporary directory for ETL
+    boost::filesystem::create_directory("temp");
```
Comment (yperbasis) [inline db/silkworm/etl/collector_test.cpp:52]: Better use TemporaryDirectory again. TemporaryDirectory creates a fresh unique temporary dir and automatically cleans it up.

File: db/silkworm/etl/collector_test.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        CHECK(value->compare(entry.value) == 0);
+    }
+    // Check wheter temporary files were cleaned
+    for (size_t i = 0; i < 10; i++) {
+        fs::path path{fs::path("temp") / fs::path("tmp-" + std::to_string(i))};
+        CHECK(!fs::exists(path));
+    }
+}
+
+TEST_CASE("collect_and_load") {
```
Comment (yperbasis) [inline db/silkworm/etl/collector_test.cpp:81]: Code duplication! Is transform function the only difference between these two tests? Extract a common function that takes transform function as an argument and call it twice with different transform functions.

**Review by yperbasis — CHANGES_REQUESTED** [review]

File: db/silkworm/etl/collector_test.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    // Initialize temporary Database
+    lmdb::DatabaseConfig db_config{tmp_dir.path(), 32 * kMebi};
+    db_config.set_readonly(false);
+    auto env{lmdb::get_env(db_config)};
+    auto txn{env->begin_rw_transaction()};
+    // Create temporary directory for ETL
+    boost::filesystem::create_directory("temp");
+    // Generate Test Entries
+    auto set{generate_entry_set(1000)};         // 1000 entries in total
+    auto collector{Collector("temp", 1600)};    // 100 entries per file
```
Comment (yperbasis) [inline db/silkworm/etl/collector_test.cpp:55]:
```suggestion
    auto collector{Collector("temp", 16 * 100)};    // 100 entries per file (each entry is 16 bytes)
```

**Review by yperbasis — APPROVED** [review]

**Pull request merged**